crates/ruff_linter/src/linter.rs

```diff
@@ -329,6 +326,14 @@ pub fn check_path(
         }
     }
 
+    // Apply range suppressions before noqa, so that #noqa on a line already covered by a range
+    // will be reported as unused noqa rather than unused range.
+    // TODO: check if enabled?
+    {
+        let suppressions = Suppressions::from_tokens(locator.contents(), tokens);
+        suppressions.filter_diagnostics(&mut context);
+    }
+
     // Enforce `noqa` directives.
     if noqa.is_enabled()
```

Comment:
I think we should only apply the filtering if `noqa_is_enabled`. I'm not a 100% sure but I think this is more a terminology issue. The CLI has a `--ignore-noqa` comment which you can use to show all diagnostics even if noqa's are disabled (https://github.com/astral-sh/ruff/issues/3070). I think we want this flag to also toggle `ruff:disable`/`ruff:enable`

I also suggest reviewing the other call sites where we set `noqa::Disabled` to see if that should apply to `ruff:disable` as well